tox.ini

```diff
@@ -1,61 +1,51 @@
 [tox]
 envlist = tests
-skipsdist = true
-minversion = 3.8.0
+minversion = 3.25.0
 requires =
-  tox-pip-sync
-  tox-pyenv
-  tox-envfile
-  tox-run-command
-tox_pyenv_fallback = false
+    tox-envfile
+    tox-faster
+    tox-run-command
+    tox-recreate
+isolated_build = true
 
 [testenv]
-skip_install = true
-sitepackages = {env:SITE_PACKAGES:false}
-setenv = dev: PYTHONPATH = .
+skip_install =
+    format,checkformatting,coverage,template: true
+setenv =
+    PYTHONUNBUFFERED = 1
+    OBJC_DISABLE_INITIALIZE_FORK_SAFETY = YES
 passenv =
     HOME
-    EXTRA_DEPS
-    dist: BUILD
+    PYTEST_ADDOPTS
 deps =
-    {tests,lint}: -e .[tests]
+    dev: ipython
+    format,checkformatting: black
+    format,checkformatting: isort
+    lint: toml
     lint: pylint
     lint: pydocstyle
-    {checkformatting,format}: black
-    {checkformatting,format}: isort[requirements]
-    coverage: coverage
-    {lint,release,initialrelease}: packaging
-    {dist,initialrelease}: twine
-    {dist,initialrelease}: wheel
-    {env:EXTRA_DEPS:}
-whitelist_externals =
-    hdev
-    release: {toxinidir}/bin/check_that_git_working_tree_is_clean.sh
-    {dist,initialrelease}: rm
-    {release,initialrelease}: sh
-    {release,initialrelease}: git
+    lint: pycodestyle
+    lint,tests: pytest-mock
+    lint,tests,coverage: coverage[toml]
+    lint,tests,functests: pytest
+    lint,tests,functests: factory-boy
+    lint,tests,functests: h-matchers
+    lint,template: cookiecutter
+    functests,lint: webtest
+depends =
+    coverage: tests,py{38}-tests
 commands =
-    tests: hdev config --if tool.hdev.tox.test_extras --run --template "pip install --quiet --use-feature=in-tree-build \{0\}"
-    tests: coverage run -m pytest tests
-
-    lint: pydocstyle --explain src
-    lint: pydocstyle --config tests/.pydocstyle --explain tests
-    lint: pylint {posargs:src bin}
-    lint: pylint --rcfile=tests/.pylintrc tests
-
-    format: black {posargs:src tests bin}
-    format: isort --atomic .
-
-    checkformatting: black --check {posargs:src tests bin}
-    checkformatting: isort --quiet --check-only .
-
+    dev: {posargs:ipython --classic --no-banner --no-confirm-exit}
+    format: black src tests bin
+    format: isort --atomic src tests bin
+    checkformatting: black --check src tests bin
+    checkformatting: isort --quiet --check-only src tests bin
+    lint: pylint src bin
+    lint: pylint --rcfile=tests/pyproject.toml tests
+    lint: pydocstyle src tests bin
+    lint: pycodestyle src tests bin
+    tests: coverage run -m pytest --failed-first --new-first --no-header --quiet {posargs:tests/unit/}
+    functests: pytest --failed-first --new-first --no-header --quiet {posargs:tests/functional/}
```

Comment:
Looking at this part of the diff, putting some space between commands is easier on the eyeballs.